src/tests/functional.spec.jsx

```diff
@@ -0,0 +1,173 @@
+import 'react-dom'
+import React from 'react'
+import {stripIndent} from 'common-tags'
+import {mount} from 'enzyme'
+
+import {
+  getCss,
+  removeWhitespace
+} from './utils'
+
+import CreateApp from './App'
+
+let Styled
+let styled
+
+const mockNameGenerators = () => {
+  let styledCounter = 0
+  let jssCounter = 0
+
+  jest.mock('../utils/generateTagName')
+  jest.mock('jss/lib/utils/generateClassName')
+
+  const generateTagName = require('../utils/generateTagName').default
+  const generateClassName = require('jss/lib/utils/generateClassName').default
+
+  // $FlowIgnore
+  generateTagName.mockImplementation((tagName: string) => `${tagName}-${++styledCounter}`)
+  generateClassName.mockImplementation(rule => `${rule.name}-${++jssCounter}`)
```

Comment:
counter is important for styled-jss, because it shows an ordering of mounting, but for jss it may be static really